test/axes/numericAxisTests.ts

```diff
@@ -431,4 +431,36 @@ describe("NumericAxis", () => {
     svg.remove();
   });
 
+  it("reasonably approximates tick label sizes with approximate measuring", () => {
+    var SVG_WIDTH = 500;
+    var SVG_HEIGHT = 100;
+
+    [[-1, 1],
+     [0, 10],
+     [0, 999999999]
+    ].forEach((domainBounds) => {
+      var scale = new Plottable.Scales.Linear();
+      scale.domain(domainBounds);
+
+      var svgApprox = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var numericAxisApprox = new Plottable.Axes.Numeric(scale, "left");
+      numericAxisApprox.usesTextWidthApproximation(true);
+      numericAxisApprox.renderTo(svgApprox);
+      var widthApprox = (<any> numericAxisApprox)._computeWidth();
+
+      var svgExact = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
```

Comment:
It should totally work.  To keep things more concise, I don't think you even need to use two numeric axes.  You just need to render one with approximation and then switch if it uses TextWidthApproximation.